app/src/org/commcare/dalvik/activities/CommCareSetupActivity.java

```diff
@@ -550,22 +556,108 @@ protected void deliverError(CommCareSetupActivity receiver,
     public boolean onCreateOptionsMenu(Menu menu) {
         super.onCreateOptionsMenu(menu);
         menu.add(0, MODE_ARCHIVE, 0, Localization.get("menu.archive")).setIcon(android.R.drawable.ic_menu_upload);
+        menu.add(0, MODE_SMS, 1, Localization.get("menu.sms")).setIcon(android.R.drawable.stat_notify_chat);
         return true;
     }
 
     @Override
     public boolean onPrepareOptionsMenu(Menu menu) {
         super.onPrepareOptionsMenu(menu);
-        
+
         return true;
     }
-    
+
+    /**
+     * Scan SMS messages for texts with profile references.
+     * @param installAutomatically install automatically if reference is found
+     */
+    private void performSMSInstall(boolean installAutomatically){
+        String profileLink = null;
+        try {
+            profileLink = this.scanSMSLinks();
+        } catch(SignatureException e){
+            // possibly we want to do something more severe here? could be malicious
+            e.printStackTrace();
+            Toast.makeText(this, Localization.get("menu.sms.not.found"), Toast.LENGTH_LONG).show();
+        }
+        if (profileLink != null) {
+            // we found a valid profile link, either start install automatically
+            // or move to READY_TO_INSTALL state
+            Log.v("install", "Performing SMS install with link : " + incomingRef);
+            incomingRef = profileLink;
+            if (installAutomatically) {
+                startResourceInstall();
+            } else {
+                uiState = UiState.READY_TO_INSTALL;
+                Toast.makeText(this, Localization.get("menu.sms.ready"), Toast.LENGTH_LONG).show();

```

Comment:
@gcapalbo @wpride commented below but I'll add it inline here - we shouldn't worry about this (especially if it's annoying) because presumably the user will be in CommCare at this point.  If they delete the text before completing the workflow, they probably aren't super interested anyway.